### ISSUE TITLE
commend_train.sh

### ISSUE THREAD

**Issue by sqdy2669** [rise-policy/RISE#4]
In the process of reproduction, I encountered some difficulties, and the model's generalization performance is relatively poor. Could I obtain your `commend_train.sh` configurations for different tasks? I would like to modify our task parameters based on your configuration settings for various tasks.

**Comment by sqdy2669** [comment]
我在复现的过程中遇到了些困难，模型的泛化性比较差，我可不可以获取您在不同任务中的commedn_train.sh配置。我想根据您对不同任务的配置参数来对我们的任务进行修改

**Comment by chenxi-wang** [comment]
Hi, we employ the same training config for all the tasks. It's hard to identify your problem. Could you provide more details？For example, what type of tasks are you working on? How many demos are you using? Have you check the data by visualization (see readme for details)? Have you ever tested other baselines with your data?

Usually the final loss should be less than 0.005. If your final loss is obviously larger than this, you may check the input point cloud and gt trajectories, or train the policy for more iterations. I hope it helps.

**Comment by chenxi-wang** [comment]
The trajectories are not contiguous, is it collected by keyboard? We recommend using dense trajectories.

Btw, it seems like using dexhand as end effector. Does tcp align to the flange pose? The keyposes visualized seems not starting from a location tightly associated with the arm.

**Comment by sqdy2669** [comment]
Thanks for your suggestions. Since our dataset was collected via VR teleoperation, the TCP (Tool Center Point) data may appear scattered.

**Comment by chenxi-wang** [comment]
Sorry I mistake the gripper for a hand lol. You can collect tcp data using an independent thread, and you can get the high-freq trajectories.

**Comment by Galaxies99** [comment]
Hi,

VR-teleoperation should be okay. What is the data collection frequency? From your visualization, it seems that the tcp between two consecutive steps is a little scattered during free moving (trajectories on the top of the yellow basket), while the difference between two consecutive steps is dense during operation (trajectory near the yellow basket). This inconsistency may cause inferior performance.

For our data collection, we record the TCP data at 10Hz, and we sample them using differences of translation, rotation, and gripper width to remove redundant actions. For the action at two adjacent timesteps, if all the differences are less than the thresholds, only the first action is retained. The threshold for translation and gripper width is 5mm, and the rotation threshold is pi/24. This can ensure consistent robot trajectories and facilitate policy learning.

Also, there are some checkboxes that you can check yourselves to see if other issues have caused the performance.
- [ ] `TRANS_MIN/MAX` are correctly set in `dataset/constants.py`;
- [ ] `WORKSPACE_MIN/MAX` are correctly set in `dataset/constants.py`, and the evaluation is conducted WITHIN this workspace.
- [ ] Hand-eye calibration is relatively accurate. You can also visualize the tcp of the current frame in the figure to check whether it is accurate.

Also, it would be beneficial to provide final loss, # batch per epoch, and other useful information for reproducing your results.

**Comment by Galaxies99** [comment]
Great. Now you can try to train the policy using these demonstrations. Remember to check the checkboxes above to ensure everything is working correctly. Let me know if you need assistance during policy training and deployment.

By the way, I am still curious about your "generalization" settings, specifically what aspect of "generalization" do you refer to?

**Comment by sqdy2669** [comment]
The generalization issue manifests in object generalization – when the target object is relocated to new positions, the robotic arm fails to operate correctly.

**Comment by Galaxies99** [comment]
Just a few simple checks, this "new location" should fall within the robot workspace, and the object should be visible from the point cloud. 

And do the demonstrations contain similar object positions (not exactly the same, just similar is okay)? For example, if your demonstrations are "pick the object from the left and place it into the basket on the right", then the trained policy is difficult to learn the "pick the object from the right and place it into the basket on the left". At least several such demonstrations need to be collected to teach the policy this new behavior.

**Comment by sqdy2669** [comment]
Thanks for your response. In my tests, I found that the model produces entirely different paths under the same environment, with substantial variations. What might be causing this? Is it possible that too many training steps led to overfitting?

**Comment by chenxi-wang** [comment]
If your trained model cannot even reproduce the actions when all the settings are almost the same as the collected data, that is not an over-fitting issue, and you may check the deployment config or camera calibration. You can visualize the predicted actions during dwployment.

For a auick check, how many demonstrations are you using, and how large is the final loss?

**Comment by sqdy2669** [comment]
Used 20 demonstration trajectories with a final loss of 0.007.

**Comment by chenxi-wang** [comment]
The loss is near the normal range, at least the model should output a roughly reasonable trajectories. You may check the deployment, and whether the objects are far from the distributed of collected data.

Btw, training policy on 20 demos is usually insufficient. We recommend 50+ demos.

**Comment by sqdy2669** [comment]
Thank you for your suggestions. We will continue to follow up and make improvements.
